components/automate-cli/pkg/verifyserver/server/api/v1/externalpostgresql_test.go

```diff
@@ -0,0 +1,138 @@
+	ExternalPgfailureMessage          = "{\"status\":\"SUCCESS\",\"result\":{\"passed\":false,\"checks\":[{\"title\":\"External Postgresql Connection failed\",\"passed\":false,\"status\":\"\",\"success_msg\":\"\",\"error_msg\":\"Machine is unable to connect with External Managed Postgresql\",\"resolution_msg\":\"Ensure that the Postgres configuration provided is correct\"}]}}"
+	ExternalPgInternalServerError     = "{\"status\":\"FAILED\",\"result\":null,\"error\":{\"code\":500,\"message\":\"Internal Server Error\"}}"
+	ErrorBodyParser                   = "{\"status\":\"FAILED\",\"result\":null,\"error\":{\"code\":400,\"message\":\"invalid character '}' looking for beginning of object key string\"}}"
+)
+
+func SetupMockExternalPostgresqlService(responseBody *models.ExternalPgResponse, err error) externalpostgresqlservice.ISExternalPostgresqlService {
+	return &externalpostgresqlservice.MockExternalPostgresqlService{
+		GetPgConnectionFunc: func(*models.ExternalPgRequest) (*models.ExternalPgResponse, error) {
+			return responseBody, err
+		},
+	}
+}
+
+func SetupExternalPostgresqlHandlers(pg externalpostgresqlservice.ISExternalPostgresqlService) (*fiber.App, error) {
+	log, _ := logger.NewLogger("text", "debug")
+	fconf := fiber.Config{
+		ServerHeader: server.SERVICE,
+		ErrorHandler: fiberutils.CustomErrorHandler,
+	}
+	app := fiber.New(fconf)
+	handler := v1.NewHandler(log).
+		AddExternalPostgresqlService(pg)
+	vs := &server.VerifyServer{
+		Port:    server.DEFAULT_PORT,
+		Log:     log,
+		App:     app,
+		Handler: handler,
+	}
+	vs.SetupRoutes()
+	return vs.App, nil
+}
+
+func TestExternalPostgresql(t *testing.T) {
+	tests := []struct {
+		description   string
+		responseBody  *models.ExternalPgResponse
+		expectedCode  int
+		expectedBody  string
+		expectedError error
+		body          string
+	}{
+		{
+			description: "200:success status route",
+			responseBody: &models.ExternalPgResponse{
+				Passed: true,
+				Checks: []models.ExternalPgConnectionDetails{
+					{
+						Title:         ExternalPgSuccessConnectionTitle,
+						Passed:        true,
+						SuccessMsg:    ExternalPgConnectionSuccessMsg,
+						ErrorMsg:      "",
+						ResolutionMsg: "",
+					},
+				},
+			},
+			expectedCode: 200,
+			expectedBody: ExternalPgsuccessMessage,
+			body: `{
+				"postgresql_instance_url": "A.B.C.D",
+				"postgresql_instance_port": "7432",
+				"postgresql_superuser_username": "postgres",
+				"postgresql_superuser_password": "Progress123",
+				"postgresql_dbuser_username": "postgres",
+				"postgresql_dbuser_password": "Progress123",
+				"postgresql_root_cert": "----- BEGIN CERTIFICATE ------"
+			}`,
+		},
+		{
+			description:  "400: body parser error",
+			responseBody: nil,
+			expectedCode: 400,
+			expectedBody: ErrorBodyParser,
+			body: `{
+						"postgresql_instance_url": "A.B.C.D",
+						"postgresq_instance_port": "7432",
```

Comment:
add a test for blank values in the fields

Reply:
done